src/frameworks/next/utils.ts

```diff
@@ -0,0 +1,41 @@
+import { existsSync } from "fs";
+import { join } from "path";
+import { readJSON } from "fs-extra";
+import type { ExportMarker, ImageManifest } from "./interfaces";
+
+/**
+ * Check if `/app` directory is used in the Next.js project.
+ * @param sourceDir location of the source directory
+ * @return true if app directory is used in the Next.js project
+ */
+export function usesAppDirRouter(sourceDir: string): boolean {
+  return existsSync(join(sourceDir, "app")) ? true : false;
```

Comment:
can drop the ternary
would rather we use an async API so we can parallelize down the road with reworking

though I'd prefer we do something a little more robust here. I was thinking we could [check for keys in the app-path-routes-manifest.json](https://github.com/firebase/firebase-tools/blob/0a7b583ef3d5aedc007f085fa7ecf0ed645d9777/src/frameworks/next/index.ts#L109). that said we're doing this because app dir is unstable... hmmmm, maybe we just drop the ternary and presence of app dir at all trips

Reply:
Agreed, the ternary wasn't necessary since `existsSync` returns a boolean. I am checking the `app-path-routes-manifest.json` now.